dali/pipeline/pipeline.h

```diff
@@ -394,10 +394,13 @@ class DLL_PUBLIC Pipeline {
   DLL_PUBLIC void SaveGraphToDotFile(const std::string &filename, bool show_tensors = false,
                                      bool show_ids = false, bool use_colors = false);
 
+  /// @{
   /**
-   * @brief Returns the batch size that will be produced by the pipeline.
+   * @brief Returns the maximum batch size that can be processed by the Pipeline
    */
-  DLL_PUBLIC inline int batch_size() const { return batch_size_; }
+  DLL_PUBLIC inline int batch_size() const { return max_batch_size_; }
+  DLL_PUBLIC inline int max_batch_size() const { return max_batch_size_; }
```

Comment:
Pipeline::SerializeToProtobuf() should probably use this call (`max_batch_size`) for `pipe.set_batch_size(...)` instead of `this->batch_size()`.